crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -137,7 +143,31 @@ impl Trade {
     ///
     /// Denominated in NATIVE token
     fn native_surplus(&self, prices: &auction::Prices) -> Result<eth::Ether, Error> {
-        let surplus = self.surplus().ok_or(Error::Surplus(self.sell, self.buy))?;
+        let limit = match self.policies.first() {
+            Some(order::FeePolicy::PriceImprovement {
+                factor: _,
+                max_volume_factor: _,
+                quote,
+            }) => adjust_quote_to_order_limits(
+                fee::Order {
+                    sell_amount: self.sell.amount.0,
+                    buy_amount: self.buy.amount.0,
+                    side: self.side,
+                },
+                fee::Quote {
+                    sell_amount: quote.sell.amount.0,
+                    buy_amount: quote.buy.amount.0,
+                    fee_amount: quote.fee.amount.0,
+                },
+            )?,
+            _ => Amounts {
+                sell: self.sell.amount,
+                buy: self.buy.amount,
+            },
+        };
```

Comment:
This still needs to be updated, will mark the review as change required because of this.